File: src/TypeFormatter/HiddenTypeFormatter.ts
import type { Definition } from "../Schema/Definition.js";
import type { SubTypeFormatter } from "../SubTypeFormatter.js";
import type { BaseType } from "../Type/BaseType.js";
import { HiddenType } from "../Type/HiddenType.js";

export class HiddenTypeFormatter implements SubTypeFormatter {
    public supportsType(type: BaseType): boolean {
        return type instanceof HiddenType;
    }
    public getDefinition(type: HiddenType): Definition {
        return { additionalProperties: false };
    }
    public getChildren(type: HiddenType): BaseType[] {
        return [];
    }
}

File: internal/formatter/hidden_type_formatter.go
package formatter

import (
	"github.com/vega/ts-json-schema-generator/internal/schema"
	"github.com/vega/ts-json-schema-generator/internal/types"
)

// HiddenTypeFormatter mirrors src/TypeFormatter/HiddenTypeFormatter.ts.
type HiddenTypeFormatter struct{}

func NewHiddenTypeFormatter() *HiddenTypeFormatter { return &HiddenTypeFormatter{} }

func (f *HiddenTypeFormatter) SupportsType(t types.Type) bool {
	_, ok := t.(*types.HiddenType)
	return ok
}

func (f *HiddenTypeFormatter) GetDefinition(t types.Type) *schema.Definition {
	return &schema.Definition{AdditionalProperties: false}
}

func (f *HiddenTypeFormatter) GetChildren(t types.Type) []types.Type { return nil }
